segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeepertier2/StorageLedger.java

```diff
@@ -0,0 +1,183 @@
+        if (older != null) {
+            this.length -= older.getLh().getLength();
+        }
+        this.length += ledgerData.getLh().getLength();
+    }
+
+    /**
+     * Increase length of the StorageLedger as a side effect of the write operation.
+     * This is just a cache operation. The length is not persisted.
+     *
+     * @param size size of data written.
+     */
+    public synchronized void increaseLengthBy(int size) {
+        this.length += size;
+    }
+
+
+    public synchronized CompletableFuture<Void> deleteAllLedgers() {
+        return CompletableFuture.allOf(
+                this.dataMap.entrySet().stream().map(entry -> manager.deleteLedger(entry.getValue().getLh())).toArray(CompletableFuture[]::new));
+    }
+
+    public LedgerData getLastLedgerData() {
+        if (this.dataMap.isEmpty()) {
+            return null;
+        } else {
+            return this.dataMap.entrySet().stream().max((entry1, entry2) -> entry1.getKey() - entry2.getKey()).get().getValue();
+        }
+    }
+
+    public void setSealed() {
+        sealed = true;
+    }
+
+    /**
+     * Creates a list of curator transaction for merging source StorageLedger in to this.
+     * @param source Name of the source ledger.
+     * @return list of curator operations.
+     */
+    public List<CuratorOp> addLedgerDataFrom(StorageLedger source) {
+        List<CuratorOp> retVal = source.dataMap.entrySet().stream().map(entry -> {
+            int newKey = entry.getKey() + this.length;
+            this.dataMap.put(newKey, entry.getValue());
+            return manager.createAddOp(this.name, newKey, entry.getValue());
+        }).collect(Collectors.toList());
+
+        this.length  += source.length;
+        return retVal;
+    }
+
+    public CompletableFuture<LedgerData> getLedgerDataForReadAt(long offset) {
+        CompletableFuture<LedgerData> retVal = new CompletableFuture<>();
+        if (offset >= length) {
+            retVal.completeExceptionally(new BadOffsetException(this.getName(), length, offset));
+        }
+        Optional<Map.Entry<Integer, LedgerData>> found = dataMap.entrySet().stream().filter(entry -> (entry.getKey() <= offset) && (offset < (entry.getKey() + entry.getValue().getLength()))).findFirst();
+        if (found.isPresent()) {
+            retVal.complete(found.get().getValue());
+        } else {
+            retVal.completeExceptionally(new BadOffsetException(this.getName(), length, offset));
+        }
+        return retVal;
+    }
+
+    public static StorageLedger deserialize(StorageLedgerManager manager, String segmentName, byte[] bytes, int version) {
+        ByteBuffer bb = ByteBuffer.wrap(bytes);
+        return new StorageLedger(manager, segmentName, version, bb.getLong(), false);
+    }
+
+    public void setContainerEpoc(long containerEpoc) {
+        this.containerEpoc = containerEpoc;
+    }
+
+    public byte[] serialize() {
+        int size = Long.SIZE;
```

Comment:
Using `BitConverter` is faster than using `ByteBuffer` if all you need to write is ints or longs